src/app/store/Cart/Cart.reducer.js

```diff
@@ -31,9 +32,26 @@ const updateCartTotals = (action) => {
 
 const updateAllProductsInCart = (action) => {
     const { products } = action;
+    const parameteredProducts = Object.keys(products).reduce((accum, key) => {
+        const currentItem = products[key];
+        if (currentItem.type_id !== 'simple') {
+            const { variants, configurable_options, configurableVariantIndex } = currentItem;
+            const selectedVariant = variants[configurableVariantIndex].product;
+            const required_params = configurable_options.map(({ attribute_code }) => attribute_code);
+            const parameters = generateParameters(selectedVariant.attributes, required_params);
+            Object.assign(selectedVariant, { parameters });
```

Comment:
I would instead do `variants[configurableVariantIndex].product = { ... selectedVariant, parameters }`. Because otherwise it violates functional programming paradigm.

Reply:
Could you also explain necessity of this operation?